airflow/contrib/hooks/gcp_vision_hook.py

```diff
@@ -0,0 +1,254 @@
+        response = self._handle_request(lambda **kwargs: client.get_product(**kwargs), name=name, retry=retry,
+                                        timeout=timeout, metadata=metadata)
+        self.log.info('Product retrieved:\n%s', response)
+        return MessageToJson(response)
+
+    @GoogleCloudBaseHook.fallback_to_default_project_id
+    def update_product(self, product, location=None, product_id=None, update_mask=None, project_id=None,
+                       retry=None, timeout=None, metadata=None):
+        """
+        For the documentation see:
+        :py:class:`~airflow.contrib.operators.gcp_vision_operator.CloudVisionProductUpdateOperator`
+        """
+        client = self.get_client()
+        product = self._determine_product_name(product, product_id, location, project_id)
+        self.log.info('Updating ProductSet: %s', product.name)
+        response = self._handle_request(lambda **kwargs: client.update_product(**kwargs), product=product,
+                                        update_mask=update_mask, retry=retry, timeout=timeout,
+                                        metadata=metadata)
+        self.log.info('Product updated:\n%s', response)
+        return MessageToJson(response)
+
+    @GoogleCloudBaseHook.fallback_to_default_project_id
+    def delete_product(self, location, product_id, project_id=None, retry=None, timeout=None, metadata=None):
+        """
+        For the documentation see:
+        :py:class:`~airflow.contrib.operators.gcp_vision_operator.CloudVisionProductDeleteOperator`
+        """
+        client = self.get_client()
+        name = PRODUCT_NAME_TEMPLATE.format(project_id, location, product_id)
+        self.log.info('Deleting ProductSet: %s', name)
+        response = self._handle_request(lambda **kwargs: client.delete_product(**kwargs), name=name,
+                                        retry=retry, timeout=timeout, metadata=metadata)
+        self.log.info('Product with the name [%s] deleted:', name)
+        return response
+
+    def _handle_request(self, fun, **kwargs):
+        try:
+            return fun(**kwargs)
+        except GoogleAPICallError as e:
+            if isinstance(e, AlreadyExists):
+                raise e
+            else:
+                self.log.error('The request failed, details below:')
+                raise AirflowException(e)
+        except RetryError as e:
+            self.log.error('The request failed due to a retryable error and retry '
+                           'attempts failed.')
+            raise AirflowException(e)
+        except ValueError as e:
+            self.log.error('The request failed, the parameters are invalid.')
+            raise AirflowException(e)
+
+    def _get_autogenerated_id(self, response):
+        try:
+            name = response.name
+        except AttributeError as e:
+            raise AirflowException('Unable to get name from response... [{}]\n{}'.format(response, e))
+        self.log.info('name: ' + name)
```

Comment:
```suggestion
        self.log.info('name: %s', name)
```
I would also consider the whole form of this message.